bonsai/UseCases/BudgetGateway.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  BudgetGateway.swift
+//  bonsai
+//
+//  Created by hoang on 21.12.2021.
+//
+
+import Foundation
+import CoreData
+
+public protocol BudgetProtocol {
+   var budgetID: String { get }
+   var name: String { get }
+   var amount: Double { get }
+   var period: Int64 { get } // days ??
+}
+
+public final class BudgetGateway {
+
+   let mainContext: NSManagedObjectContext
+
+   init(mainContext: NSManagedObjectContext = DataController.sharedInstance.mainContext) {
+      self.mainContext = mainContext
+   }
+
+   @discardableResult
+   func create(name: String, amount: Double, period: Int64) throws -> Budget {
+      let budget = Budget(
+         context: mainContext,
+         name: name,
+         amount: amount,
+         period: period)
+      try mainContext.save()
+      return budget
+   }
+
+   func update(budget: Budget) throws {
+      try mainContext.save()
```

Comment:
if the `budget` was created or taken from another context, a data might not be updated